dht_net.go

```diff
@@ -56,6 +56,10 @@ func (dht *IpfsDHT) handleNewMessage(s inet.Stream) {
 	w := newBufferedDelimitedWriter(cw)
 	mPeer := s.Conn().RemotePeer()
 
+	ctx = dht.logSpan(ctx, "handleNewMessage")
+	log.SetTag(ctx, "peer", mPeer)
+	defer log.Finish(ctx)
```

Comment:
this is a very long-lived span that might actually never complete because of the loop. Is it actually useful?